src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -1050,4 +1050,21 @@ private Map<String,Object> collectLines(String response, final List<String> line
         return content;
     }
 
+    @NonNull
+    @Override
+    public SCMFile getFile(@NonNull BitbucketSCMFile file) throws IOException, InterruptedException {
+        String branchOrHash = file.getHash().contains("+") ? file.getRef() : file.getHash();
+        String url = UriTemplate.fromTemplate(this.baseURL + API_BROWSE_PATH + "{&type,blame,size")
+                .set("owner", getUserCentricOwner())
+                .set("repo", repositoryName)
+                .set("path", file.getPath().split(Operator.PATH.getSeparator()))
+                .set("at", branchOrHash)
+                .set("type", true)
+                .set("blame", false)
+                .expand();
+        String response = getRequest(url);
+        // TODO deserialize response...
```

Comment:
@KalleOlaviNiemitalo could you help me to understand the response of bitbucket server?

https://developer.atlassian.com/server/bitbucket/rest/v905/api-group-repository/#api-api-latest-projects-projectkey-repos-repositoryslug-browse-path-get

It's not clear what return. Do you have an example?

Reply:
What do you think if after I test the cloud part, than merge leaving this method as placeholder so you could provide a PR for the server part?